### PR TITLE
Feature: add Razorpay per-form settings migration step

### PR DESCRIPTION
**Pull request by glaubersilva**
Resolves [GIVE-934]

Related PR: https://github.com/impress-org/give-razorpay/pull/84

## Description

This PR adds a new form migration step to ensure that the Razorpay per-form settings will be migrated to new V3 forms.

## Affects

The form migration process.

## Visuals

image

image

## Testing Instructions

1. Set up a v2 form and setup Razorpay per-form settings;
2. Migrate the form created in the previous step to a v3 form;
3. Make sure the settings from step 1 were migrated and available in the Payment Gateways block settings.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-934]: https://stellarwp.atlassian.net/browse/GIVE-934?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/FormMigration/Steps/RazorpayPerFormSettings.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Give\FormMigration\Steps;
+
+use Give\FormMigration\Contracts\FormMigrationStep;
+
+/**
+ * @unreleased
+ */
+class RazorpayPerFormSettings extends FormMigrationStep
```
Comment (jonwaldstein) [inline src/FormMigration/Steps/RazorpayPerFormSettings.php:10]: This should have a `canHandle()` method so the attributes are only added when the v2 form is customized.  We should be checking if the meta exists before processing

File: tests/Feature/FormMigration/Steps/TestRazorpayPerFormSettings.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+use Give\FormMigration\DataTransferObjects\FormMigrationPayload;
+use Give\FormMigration\Steps\RazorpayPerFormSettings;
+use Give\Tests\TestCase;
+use Give\Tests\TestTraits\RefreshDatabase;
+use Give\Tests\Unit\DonationForms\TestTraits\LegacyDonationFormAdapter;
+
+/**
+ * @unreleased
+ */
+class TestRazorpayPerFormSettings extends TestCase
```
Comment (jonwaldstein) [inline tests/Feature/FormMigration/Steps/TestRazorpayPerFormSettings.php:14]: We'll want to test that `canHandle` method to make sure this is only processed when necessary.

**Review by jonwaldstein — CHANGES_REQUESTED** [review]

File: src/FormMigration/Steps/RazorpayPerFormSettings.php (hunk 1 above)
Comment (glaubersilva) [reply src/FormMigration/Steps/RazorpayPerFormSettings.php:10]: Resolved in 251a5e452a0de49f45e1fa0a67e1a860fb20be91

File: tests/Feature/FormMigration/Steps/TestRazorpayPerFormSettings.php (hunk 2 above)
Comment (glaubersilva) [reply tests/Feature/FormMigration/Steps/TestRazorpayPerFormSettings.php:14]: Resolved in d66bd27e5f620f475f022f4c063225107fc86176

**Comment by glaubersilva** [comment]
@jonwaldstein This is ready for re-review.

**Review by jonwaldstein — APPROVED** [review]
@glaubersilva thanks for the update, great job!

**Review by rickalday — APPROVED** [review]
Passed manual QA tests.

**Pull request merged**